src/main/java/ch/ivyteam/ivy/maven/InstallEngineMojo.java

```diff
@@ -149,23 +176,64 @@ private void handleWrongIvyVersion(ArtifactVersion installedEngineVersion) throw
 
   private void downloadAndInstallEngine(boolean cleanEngineDir) throws MojoExecutionException
   {
+    EngineDownloader engineDownloader = new EngineDownloader();
+
     if (autoInstallEngine)
     {
       getLog().info("Will automatically download Engine now.");
-      EngineDownloader engineDownloader = new EngineDownloader();
-      File downloadZip = engineDownloader.downloadEngine();
+
+
+      File downloadZip = null;
+      if(downloadUsingMaven)
```

Comment:
I'd recommend to encapsulate at least the ZIP download via maven in an extra method. 
As the execution flow that defines the installed engine is already pretty complex.

Reply:
Sure, i'll refactor the PR (first draft was mostly intendend as demonstration of a possible solution) - i think it would be better to encapsulate the downloading of the artifact with maven inside the EngineDownloader, too. I'll push a new draft soon.